src/README.md

```diff
@@ -0,0 +1,55 @@
+
+<hr>
+
+### Layered Architecture Description (Tiered Approach):
+
+1. **Layer 1 (L1 - HTTP Client Wrapper):**
+
+    - **Purpose**: This L1 layer serves as the foundational layer that wraps/encapsulates around an HTTP client, specifically `httpx` for Provena Python Client. This layer allows us to abstract the direct handling of HTTP methods (GET, PUT, POST, DELETE) and centralises certain HTTP client settings such as timeouts.
+
+    - **Current Approach**: In the current implementation of Layer 1 (L1 - HTTP Client Wrapper) within the Provena Python Client, we use the `httpx` library to handle HTTP requests in an **asynchronous fashion**. This allows for better performance of the client and allows for non-blocking requests. Furthermore, each HTTP methods in our client - GET, PUT, POST, DELETE -- is designed to accept and handle the necessary parameters such as auth, headers, query params, and other body params in accordance to the Provena API requirements. In the current approach, `httpx`, is used within a context-manager which ensures that each API request is handled in a fresh session, maintaining clean and reliable session management throughout API interactions.
+
+    - **Benefit**: The benefit of having this separate HTTP layer is that it provides us with an option to replace the underlying HTTP library in the future without affecting the rest of the client library.
+    
+<hr>
+
+2. **Layer 2 (L2 - Client Interfaces):** 
+    - **Purpose:** This L2 layer serves and sits between the HTTP client wrapper (Layer 1) and the user interface module (Layer 3). It's responsible for preparing the API request payload and parsing API responses to interactive Python datatype. This layer helps us simplify and streamline the process of constructing the API requests and parsing responses within the Provena Python Client. 
+
+    - **Current Approach:** The current implementation of Layer 2 (L2 - Client Interfaces) it leverages the authentication and configuration settings that are provided by the user to securely and effectively manage the API communications. Furthermore, each API within this layer has its dedicated client that encapsulates specific business logic and functionalities relevant to that respective API. 
+
+    Currently, the L2 layer responsibilities include constructing and preparing the API request payload based on the respective API and endpoint (a defined function within the client) requirements and the payload is sent through to the HTTP Client provided by L1 through the methods of `parsed_get_request_with_status`, `parsed_get_request`, `parsed_post_request`, etc. Additionally, the L2 layer is responsible for parsing and validating the API responses returned from L1 against Pydantic models sourced from the Provena Interfaces into structured and interactive Python datatypes if successful, else it handles and raises its custom exceptions accordingly.
```

Comment:
Is this meant to also be a dot point?

Reply:
Ah yep, it is meant to be a bit more indented. I have fixed that.